forte/data/extractor/seqtagging_extractor.py

```diff
@@ -0,0 +1,187 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file implements BioSeqTaggingExtractor, which is used to extract feature
+from the tagging label.
+"""
+import logging
+from typing import List, Dict, Union
+from ft.onto.base_ontology import Annotation, EntityMention
+from forte.common.configuration import Config
+from forte.data.data_pack import DataPack
+from forte.data.converter.feature import Feature
+from forte.data.extractor.utils import bio_tagging
+from forte.data.extractor.base_extractor import BaseExtractor
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "BioSeqTaggingExtractor"
+]
+
+
+class BioSeqTaggingExtractor(BaseExtractor):
+    r"""BioSeqTaggingExtractor will the feature by performing BIO encoding
+    for the attribute of entry, aligining to the based_on entry.
+
+    Args:
+        config: An instance of `Dict` or
+            :class:`forte.common.configuration.Config`
+    """
+    def __init__(self, config: Union[Dict, Config]):
+        super().__init__(config)
+
+        if self.config.attribute is None:
+            raise AttributeError("attribute is required "
+                            "in BioSeqTaggingExtractor.")
+        if self.config.based_on is None:
+            raise AttributeError("based_on is required in "
+                                "BioSeqTaggingExtractor.")
+
+    @classmethod
+    def default_configs(cls):
+        r"""Returns a dictionary of default hyper-parameters.
+
+        attribute (str): Required. The attribute name of the
+            entry from which labels are extracted.
+
+        based_on (Type[Entry]): Required. The tagging label
+            will align to the based_on Entry.
+        """
+        config = super().default_configs()
+        config.update({"attribute": None,
+                        "based_on": None})
```

Comment:
The name `based_on` is a bit confusing. Maybe we should find a more informative name, maybe `tagging_unit` or `tagging_target`. (You can also use a better name if you can come up with one.

Reply:
Got it. I will update this latter, after we solve the implementation.